### ISSUE TITLE
[BUG] Open files issue with gitfs (pygit2) Backend

### ISSUE THREAD

**Issue by Darkentik** [saltstack/salt#58577]
--> related to: https://github.com/saltstack/salt/issues/22649
**Description**
Often when we trigger manually the saltmaster to update his remote pillar from git (salt-run git_pillar.update) we got the error:
`~/ $ salt-run git_pillar.update && salt-run fileserver.update
[ERROR   ] Error occurred fetching git_pillar remote 'gitolite3@git.test.le.fp.lan:salt.git': could not open '/var/cache/salt/master/git_pillar/test/.git/objects/f2/b70f2197be8416d341162edd6c8b71f17819d6': Too many open files
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/salt/utils/gitfs.py", line 1745, in _fetch
    fetch_results = origin.fetch(**fetch_kwargs)
  File "/usr/lib/python3/dist-packages/pygit2/remote.py", line 411, in fetch
    check_error(err)
  File "/usr/lib/python3/dist-packages/pygit2/errors.py", line 64, in check_error
    raise GitError(message)
_pygit2.GitError: could not open '/var/cache/salt/master/git_pillar/test/.git/objects/f2/b70f2197be8416d341162edd6c8b71f17819d6': Too many open files`

**Setup**
/etc/salt/master.de/git.conf:
```gitfs_provider: pygit2
gitfs_base: master
git_pillar_base: master

fileserver_backend:
  - git

top_file_merging_strategy: same

gitfs_remotes:
  - gitolite3@git.test.localdomain:salt.git:
    - privkey: /PATH/key
    - pubkey: /PATH/key.pub
    - passphrase: '123456'
    - root: states
    - saltenv:
      - test:
        - ref: master

ext_pillar:
  - git:
    - gitolite3@git.test.localdomain:salt.git:
      - name: test
      - privkey: /PATH/key
      - pubkey: /PATH/key.pub
      - passphrase: '123456'
      - root: pillar
      - env: test```

**Steps to Reproduce the behavior**
Trigger manually the update of external pillar from git with:
`salt-run git_pillar.update`

**Expected behavior**
```~/ $ salt-run git_pillar.update && salt-run fileserver.update
gitolite3@git.test.localdomain:salt.git:
    None
[INFO    ] Runner completed: 20200929085114736083
```

**Versions Report**
<details><summary>salt --versions-report</summary>
(Provided by running salt --versions-report. Please also mention any differences in master/minion versions.) 

```
 $ salt --versions-report
Salt Version:
           Salt: 2019.2.5
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.7.3
      docker-py: Not Installed
          gitdb: 2.0.5
      gitpython: 2.1.11
          ioflo: Not Installed
         Jinja2: 2.10
        libgit2: 0.27.7
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 1.0.7
   msgpack-pure: Not Installed
 msgpack-python: 0.5.6
   mysql-python: Not Installed
      pycparser: 2.19
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: 0.27.4
         Python: 3.7.3 (default, Jul 25 2020, 13:03:44)
   python-gnupg: 0.4.4
         PyYAML: 3.13
          PyZMQ: 17.1.2
           RAET: Not Installed
          smmap: 2.0.5
        timelib: Not Installed
        Tornado: 4.5.3
            ZMQ: 4.3.1
 
System Versions:
           dist: debian 10.5 
         locale: UTF-8
        machine: x86_64
        release: 4.19.0-10-amd64
         system: Linux
        version: debian 10.5 
```
</details>

**Comment by OrangeDog** [comment]
What is the current file limit for salt-master? It should be set to ~100000.

**Comment by sagetherage** [comment]
@Darkentik did you set the file limit on the master? are you still seeing this issue?

**Comment by Darkentik** [comment]
Hi @OrangeDog and @sagetherage,
sorry for the delay!!!!!! I'm very busy in the moment.
@OrangeDog We started with the default value for the max_open_files and didn't set anything manually.
After we set 100000 (which is the default value) manually the error was gone.

I can't say more because i have left the company and in my new company our saltstack environment isn't that big in the moment as in the company before.

So we could close the issue yet and i will give feedback if we have the same issue with pygit2 here at the new company while our environment is growing.

Thanks guys!

Best regards,
Darkentik
